e2e-tests/app_test.go

```diff
@@ -157,28 +154,20 @@ func (l testRunLogger) RunStderr(r *Run, line []byte) {
 	l.log.Log(string(line))
 }
 
-func proxyTcp(ctx context.Context, ln net.Listener, client *yamux.Session) {
-	for ctx.Err() == nil {
-		conn, err := ln.Accept()
-		if err != nil {
-			if strings.Contains(err.Error(), "use of closed network connection") {
-				return
-			}
-
-			fmt.Printf("unable to accept connection: %+v", err)
-			continue
-		}
+func startProxy(ctx context.Context, ln net.Listener, p *Proc) {
+	srv := &http.Server{
+		Handler: http.HandlerFunc(p.ProxyReq),
+	}
+	go func() {
+		<-ctx.Done()
+		srv.Close()
+	}()
 
-		clientConn, err := client.Open()
-		if err != nil {
-			fmt.Printf("unable to open connection to running app: %+v", err)
-			_ = conn.Close()
-			continue
-		}
+	go srv.Serve(ln)
 
-		go io.Copy(conn, clientConn)
-		go io.Copy(clientConn, conn)
-	}
+	// So we're going to sleep here to give the server a chance to start up.
+	// as otherwise we see the application startup before this proxy is ready
+	time.Sleep(1 * time.Second)
```

Comment:
Is there a better solution than this? What exactly is the impact of this (i.e. what does it delay)?

Reply:
Without it we have flacky end to end tests. Normally at least one of them fails. 

This is because the proxy starts _after_ the test try’s to call the endpoint.

Reply:
I couldn’t think of a bretter way to solve it today